src/network/node.rs

```diff
@@ -206,8 +206,17 @@ fn extract_peer_id_from_multiaddr(
 
 #[cfg(test)]
 mod tests {
+    use async_std::task;
 
-    // TODO
-    #[test]
-    fn it_works() {}
+    use super::*;
+
+    #[async_std::test]
+    async fn it_works() {
+        let n1 = Node::new(None);
+        let n1_listeners = n1.swarm.listeners().last().unwrap().clone();
+        task::spawn(n1.run());
+
+        let n2 = Node::new(Some(vec![n1_listeners]));
+        task::block_on(n2.run());
+    }
```

Comment:
a few things here

- line 216 you will probably have to fix since the listeners will likely be empty and the test will panic. try waiting for `n1` to be running before getting the listeners.
- `#[async_std::test]` allows for `async` test functions